core/src/SYCL/Kokkos_SYCL.hpp

```diff
@@ -182,6 +182,31 @@ std::vector<SYCL> partition_space(const SYCL& sycl_space,
         sycl::queue(context, device, sycl::property::queue::in_order()));
   return instances;
 }
+
+namespace Impl {
+inline std::vector<sycl::device> get_sycl_devices() {
+  std::vector<sycl::device> gpu_devices =
+      sycl::device::get_devices(sycl::info::device_type::gpu);
+#if defined(KOKKOS_ARCH_INTEL_GPU) || defined(KOKKOS_IMPL_ARCH_NVIDIA_GPU) || \
+    defined(KOKKOS_ARCH_AMD_GPU)
+#if defined(KOKKOS_ARCH_INTEL_GPU)
+  sycl::backend backend = sycl::backend::ext_oneapi_level_zero;
+#elif defined(KOKKOS_IMPL_ARCH_NVIDIA_GPU)
+  sycl::backend backend = sycl::backend::ext_oneapi_cuda;
+#elif defined(KOKKOS_ARCH_AMD_GPU)
+  sycl::backend backend = sycl::backend::ext_oneapi_hip;
+#endif
+  gpu_devices.erase(std::remove_if(gpu_devices.begin(), gpu_devices.end(),
+                                   [backend](const sycl::device& d) {
+                                     return d.get_backend() != backend;
+                                   }),
+                    gpu_devices.end());
+  if (gpu_devices.empty())
+    Kokkos::abort("Error: no GPU available for execution.\n");
```

Comment:
Now I realized that this would be handled there
https://github.com/kokkos/kokkos/blob/2389f6eca377348b822224d8c270ceddfd9da5b4/core/src/impl/Kokkos_Core.cpp#L349-L353
So maybe we don't need that assertion.
Sorry I know I had told you to add but I was wrong.